File: src/main/java/com/crui/patterns/examples/turnos_medicos/after/Doctor.py
from Especialidad import Especialidad
from Turno import Turno

class Doctor:
    def __init__(self, nombre: str, especialidad: Especialidad, email: str):
        self.nombre = nombre
        self.especialidad = especialidad
        self.email = email

    def avisarCambioDeFechayHora(self, turno: Turno) -> None:
        print(
            "Mail para "
            + self.email
            + ": El turno para "
            + turno.paciente
            + " se ha cambiado a "
            + turno.fechaYHora
        )

    def __str__(self):
        return self.nombre + " (" + self.especialidad + ")"

File: src/main/java/com/crui/patterns/examples/turnos_medicos/after/Doctor.java
package com.crui.patterns.examples.turnos_medicos.after;

public class Doctor {
  String nombre;
  Especialidad especialidad;
  String email;

  public Doctor(String nombre, Especialidad especialidad, String email) {
    this.nombre = nombre;
    this.especialidad = especialidad;
    this.email = email;
  }

  public void avisarCambioDeFechayHora(Turno turno) {
    System.out.println(
        "Mail para "
            + email
            + ": El turno para "
            + turno.getPaciente()
            + " se ha cambiado a "
            + turno.getFechaYHora());
  }

  @Override
  public String toString() {
    return nombre + " (" + especialidad + ")";
  }
}
